### PR TITLE
Refactor HmGui

### LINKED ISSUE #160: [HmGui] Refactoring
- remove unsafe usage
- use proc macro for ffi code generation

### PR DESCRIPTION
**Pull request by HaronK**
Closes #160

### CODE REVIEW

File: engine/lib/phx/script/ffi_gen/Engine.lua [hunk 1]
```diff
@@ -33,6 +34,7 @@ function Loader.defineType()
         Engine = {
             Window     = libphx.Engine_Window,
             Input      = libphx.Engine_Input,
+            Hmgui      = libphx.Engine_Hmgui,
```
Comment (IllustrisJack) [inline engine/lib/phx/script/ffi_gen/Engine.lua:37]: no PascalCase? :)

File: engine/lib/phx/src/ui/hmgui/focus.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#[derive(Debug, Copy, Clone, Default, PartialEq, Eq)]
+pub enum FocusStyle {
+    #[default]
+    None,
+    Fill,
+    #[allow(dead_code)]
```
Comment (IllustrisJack) [inline engine/lib/phx/src/ui/hmgui/focus.rs:6]: this is temp i suppose

File: script/States/App/LTheory.lua [hunk 3]
```diff
@@ -111,13 +111,13 @@ function LTheory:onUpdate(dt)
     self.player:getRoot():update(dt)
     self.canvas:update(dt)
 
-    HmGui.Begin(self.resX, self.resY, InputInstance) -- required for HmGui.Draw() to work without crashing
-    HmGui.End(InputInstance)
+    Gui:beginGui(self.resX, self.resY, InputInstance) -- required for Gui:draw() to work without crashing
```
Comment (IllustrisJack) [inline script/States/App/LTheory.lua:114]: This still feels a bit weird that we have to do this. Is there no way to circumvent setting a begin and end like that?

**Review by IllustrisJack — APPROVED** [review]

File: engine/lib/phx/script/ffi_gen/Engine.lua (hunk 1 above)
Comment (HaronK) [reply engine/lib/phx/script/ffi_gen/Engine.lua]: Forgot that we can rename methods for FFI. Fixed.

File: engine/lib/phx/src/ui/hmgui/focus.rs (hunk 2 above)
Comment (HaronK) [reply engine/lib/phx/src/ui/hmgui/focus.rs]: Yes, it was temporary. Removed.

File: script/States/App/LTheory.lua (hunk 3 above)
Comment (HaronK) [reply script/States/App/LTheory.lua:114]: I'll check this when I start testing Lua part.

**Review by dgavedissian — APPROVED** [review]

**Pull request merged**